source/robot_model/src/Model.cpp

```diff
@@ -120,6 +119,30 @@ bool Model::check_collision(const state_representation::JointPositions& joint_po
     return false;
 }
 
+// Compute minimum distance
+Eigen::MatrixXd Model::compute_minimum_distance(const state_representation::JointPositions& joint_positions) {
+    Eigen::VectorXd configuration = joint_positions.get_positions();
+    pinocchio::computeDistances(this->robot_model_, this->robot_data_, this->geom_model_, this->geom_data_, configuration);
+    
+    // nb_joints is the number of joints in the robot model
+    unsigned int nb_joints = this->get_number_of_joints();
+    
+    // create a 2D Eigen vector to store the distances and initialize to zero
+    Eigen::MatrixXd distances = Eigen::MatrixXd::Zero(nb_joints, nb_joints);
+    
+    // iterate over the collision pairs and extract the distances
+    unsigned int pair_index = 0;
+    for (unsigned row_index = 0; row_index < nb_joints; ++row_index) {
```

Comment:
does this compile? shouldn't it be `unsigned int` ?

Reply:
they resolve to the same thing (I am not sure we can guarantee this for all architectures, but normally it's the same thing). It would still be better to consistently use `unsigned int` though.

Reply:
indeed they resolve to the same thing! I have updated it for consistency :)